system_metrics_collector/src/system_metrics_collector/collector.hpp

```diff
@@ -69,20 +69,21 @@ class Collector
    */
   virtual std::string GetStatusString() const;
 
-  // todo @dabonnie uptime (once start has been called)
+  // TODO(dabonnie): uptime (once start has been called)
 
-protected:
   /**
    * Start collecting data. Meant to be called after construction. Note: this locks the recursive mutex class
-   * member 'mutex'.
+   * member 'mutex'. This method is public in order for the caller to manually manage starting and
+   * stopping this collector.
    *
    * @return true if started, false if an error occurred
```

Comment:
I think this function could be a good candidate for `noexcept` since its using the return value to indicate any errors.

Reply:
That requires all the of the implemented SetupStarts to be noexcept as well, which may not be feasible. I think this is worth discussing / approaching, but out of scope here.